### ISSUE TITLE
Found a misused OpenCL API in handelocl.cpp. 

### ISSUE THREAD

**Issue by marty1885** [ROCmSoftwarePlatform/MIOpen#5]
Hello. I have been studying MIOpen's source code since it's launch. After a while, I found that MIOpen can be built on a non ROCm system with OpenCL backend (not sure if MIOpen can be built with HIP on Nvidia platforms). However building and running MIOpen with with OpenCL on a non ROCm system will fail half(5/10) of it's test with something throwing std::bad_alloc.
Digging into the source. I found the following code snippet.
 
```C++
    /* First, get the size of device list data */  
    size_t deviceListSize;  
    if(clGetContextInfo(  
           impl->context.get(), CL_CONTEXT_NUM_DEVICES, sizeof(size_t), &deviceListSize, nullptr) !=  
       CL_SUCCESS)  
    {  
        MIOPEN_THROW("Error: Getting Handle Info (device list size, clGetContextInfo)");  
    }  
  
  
    if(deviceListSize == 0)  
    {  
        MIOPEN_THROW("Error: No devices found.");  
    }  
    std::vector<cl_device_id> devices(deviceListSize);  
```
 
According to [clGetContextInfo's documentation](https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clGetContextInfo.html). When param_name is `CL_CONTEXT_NUM_DEVICES`, return type should be `cl_uint` . not `size_t`.
This causes deviceListSize being not set correctly. And thus being a large value, causing the std::vector trying to allocate really large amount of memory and thus throwing.

After changing the type of `deviceListSize` to `cl_uint` and change `sizeof(size_t)` to `sizeof(cl_uint)`. 8/10 test passes (test 5 and 6 failed, I'll open anther issue for that). 
I guess this is a compatibility issue. Will/can this be fixed on a future version of MIOpen?

Tested on: 
Manjaro Linux with Intel OpenCL SDK for CPU and Nvida OpenCL (not sure which one MIOpen uses).

**Comment by dagamayank** [comment]
@marty1885 Thanks for finding this bug. We fill fix it in the next release of MIOpen.

**Comment by dagamayank** [comment]
Fixed in 4bf6105c011a77f76943faa43a28c31b61e7e337
